File: test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/config/env/EachBeanTest.kt
package io.micronaut.docs.config.env

import io.kotest.core.spec.style.AnnotationSpec
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import io.micronaut.context.ApplicationContext
import io.micronaut.context.env.PropertySource
import io.micronaut.docs.config.env.DataSourceFactory.DataSource
import io.micronaut.inject.qualifiers.Qualifiers
import java.net.URISyntaxException

class EachBeanTest : AnnotationSpec() {

    @Test
    @Throws(URISyntaxException::class)
    fun testEachBean() {
        // tag::config[]
        val applicationContext = ApplicationContext.run(PropertySource.of(
                "test",
                mapOf(
                        "test.datasource.one.url" to "jdbc:mysql://localhost/one",
                        "test.datasource.two.url" to "jdbc:mysql://localhost/two")
        ))
        // end::config[]

        // tag::beans[]
        val beansOfType = applicationContext.getBeansOfType(DataSource::class.java)
        beansOfType.size shouldBe 2 // <1>

        val firstConfig = applicationContext.getBean(
                DataSource::class.java,
                Qualifiers.byName("one") // <2>
        )
        // end::beans[]

        firstConfig.shouldNotBeNull()

        applicationContext.close()
    }
}

File: test-suite-python/src/test/python/micronaut/docs/config/env/EachBeanTest.py
import java

from micronaut.context import ApplicationContext
from micronaut.context.env import PropertySource
from micronaut.inject.qualifiers import Qualifiers
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest
class EachBeanTest:
    @Test
    def test_each_bean(self) -> None:
        # tag::config[]
        application_context = ApplicationContext.run(PropertySource.of(
            "test",
            {
                "test.datasource.one.url": "jdbc:mysql://localhost/one",
                "test.datasource.two.url": "jdbc:mysql://localhost/two",
            },
        ))
        # end::config[]

        try:
            # tag::beans[]
            DataSource = java.type("micronaut.docs.config.env.DataSource")
            beans_of_type = application_context.getBeansOfType(DataSource)
            assert 2 == beans_of_type.size()  # <1>

            first_config = application_context.getBean(
                DataSource,
                Qualifiers.byName("one"),  # <2>
            )
            # end::beans[]

            assert first_config is not None
        finally:
            application_context.close()
